plugins/com.google.cloud.tools.eclipse.appengine.libraries.test/xsd/com.google.cloud.tools.eclipse.appengine.libraries.xsd

```diff
@@ -0,0 +1,263 @@
+   </xs:element>
+   
+   <xs:element name="extension">
+      <xs:annotation>
+         <xs:appinfo>
+            <xs:meta.element />
+         </xs:appinfo>
+      </xs:annotation>
+      <xs:complexType>
+         <xs:choice minOccurs="1" maxOccurs="unbounded">
+            <xs:element ref="library" />
+         </xs:choice>
+         <xs:attribute name="point" type="xs:string" use="required" />
+         <xs:attribute name="id" type="xs:string" />
+         <xs:attribute name="name" type="xs:string">
+            <xs:annotation>
+               <xs:appinfo>
+                  <xs:meta.attribute translatable="true" />
+               </xs:appinfo>
+            </xs:annotation>
+         </xs:attribute>
+      </xs:complexType>
+   </xs:element>
+
+   <xs:element name="library">
+      <xs:annotation>
+         <xs:documentation>
+            An App Engine library that can consist of one or more jar files with associated visibility rules and Maven coordinates.
+         </xs:documentation>
+      </xs:annotation>
+      <xs:complexType>
+         <xs:choice minOccurs="1" maxOccurs="unbounded">
+            <xs:element ref="libraryFile" />
+         </xs:choice>
+         <xs:attribute name="id" type="xs:string">
+            <xs:annotation>
+               <xs:documentation>
+                  The ID of the library. The path of the classpath container defined for this library will be 
+'com.google.cloud.tools.eclipse.appengine.libraries/ID' where ID is the value defined in this attribute. 
+The classpathContainerInitializer defined to resolve the container will have the ID 
+'com.google.cloud.tools.eclipse.appengine.libraries'. The second section of the container path (the ID after the '/') 
+will determine how to resolve the entries associated with the container.
+               </xs:documentation>
+            </xs:annotation>
+         </xs:attribute>
+         <xs:attribute name="name" type="xs:string">
+            <xs:annotation>
+               <xs:documentation>
+                  A user-friendly name for the library that can be displayed in the UI, e.g.: &quot;AppEngine Endpoints&quot;.
+               </xs:documentation>
+            </xs:annotation>
+         </xs:attribute>
+         <xs:attribute name="siteUri" type="xs:string">
+            <xs:annotation>
+               <xs:documentation>
+                  URI that can be used to display a link to the user to provide more detailed information about the library, e.g. user guides, wiki, other documentation.
+               </xs:documentation>
+            </xs:annotation>
+         </xs:attribute>
+      </xs:complexType>
+   </xs:element>
+
+   <xs:element name="libraryFile">
+      <xs:annotation>
+         <xs:documentation>
+            Describes a single jar file located in a Maven repository.
+         </xs:documentation>
+      </xs:annotation>
+      <xs:complexType>
+         <xs:sequence>
+            <xs:sequence minOccurs="0" maxOccurs="unbounded">
```

Comment:
you can simplify this by removing the sequence and pushing the minOccurs/maxOccurs attributes onto the choice